packages/fast-tooling-react/src/message-system/index.ts

```diff
@@ -0,0 +1,47 @@
+/// <reference lib="webworker" />
+
+import {
+    MessageSystemComponentTypeAction,
+    MessageSystemType,
+} from "./message-system.props";
+
+/**
+ * This is the Message System, through which:
+ * - Components may opt in/out of the messages
+ * - Data manipulation may be performed
+ *
+ * The main purpose of this is to tie together
+ * process heavy actions onto a separate thread,
+ * as well as to allow components to opt into a
+ * single source for data updates.
+ */
+
+const registeredComponents: object = {};
+
+onmessage = function(e: MessageEvent): void {
```

Comment:
no `const` / `let` / `var`? Is this some webworker syntax?

Reply:
it's a bit odd, but it's declared in the global scope see: https://github.com/microsoft/TypeScript/blob/master/lib/lib.webworker.d.ts#L5768

this is a plain javascript example but here it is implemented simply: https://github.com/mdn/simple-web-worker/blob/gh-pages/worker.js